### PR TITLE
Simplifying abstraction by Using date as indices instead of numbers

### PR DESCRIPTION
**Pull request by ivendor**
As mentioned in the title we don't need to pass for an intermediate abstractions as dates are hashable and sortable already

### CODE REVIEW

**Review by KapJI — APPROVED** [review]
Thanks! 🙂

File: tests/test_calc.py [hunk 1]
```diff
@@ -125,7 +124,7 @@ def transfer_transaction(
         1.00,  # Expected capital gain/loss
         None,
         {
-            date_to_index(datetime.date(day=1, month=5, year=2020)): {
+            (datetime.date(day=1, month=5, year=2020)): {
```
Comment (KapJI) [inline tests/test_calc.py]: No need in these brackets anymore.
```suggestion
            datetime.date(day=1, month=5, year=2020): {
```

**Comment by ivendor** [comment]
Addressed all issues and re-tested manually too

File: tests/test_calc.py (hunk 1 above)
Comment (KapJI) [reply tests/test_calc.py]: I think you missed this one. And there're more places like this below.

Comment (ivendor) [reply tests/test_calc.py]: Yep you're right, updated it now

**Pull request merged**